src/components/popup-utils.js

```diff
@@ -63,6 +63,10 @@ export const computePopupPosition = ({ side, viewport, target, element, gap }) =
   return { side: finalSide, position: finalPosition }
 }
 
+// Render popups, modals, etc. into the #modal-root element.
+// Fall back to document.body in tests.
+export const getPopupRoot = () => document.getElementById('modal-root') || document.body
```

Comment:
great general strategy here, but I would ask to not put in "for testing" fallbacks in general.  Let's have this getPopupRoot method create-if-needed the correct element with id='modal-root' to match expected behavior.  That way testing setup is the same as 'prod'

Reply:
also, I am inclined to remove the element from index.html in /public as well, to make it clear this is not a shadow dependency anymore, since the controls that need it will add-on-demand transparently through this helper.

Reply:
Updated `getPopupRoot` to create the element if it doesn't exist and removed the modal-root element from index.html.

Reply:
awesome!